server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -455,15 +455,27 @@ private void onShardFailure(final int shardIndex, @Nullable SearchShardTarget sh
             .findNext(shardIt, clusterState, e, () -> totalOps.incrementAndGet());
 
         final boolean lastShard = nextShard == null;
-        logger.debug(
-            () -> new ParameterizedMessage(
-                "{}: Failed to execute [{}] lastShard [{}]",
-                shard != null ? shard : shardIt.shardId(),
-                request,
-                lastShard
-            ),
-            e
-        );
+        if (logger.isTraceEnabled()) {
```

Comment:
The if check is not needed if you use the below logger interface
```
void debug(MessageSupplier messageSupplier);
```

Reply:
Added this check explicitly to avoid logging this twice in case trace is enabled.